bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/internal/MqttBrokerConnection.java

```diff
@@ -457,19 +458,22 @@ public void publish(String topic, byte[] payload) throws Exception {
 
                 // publish message asynchronously
                 MqttTopic mqttTopic = client.getTopic(topic);
-                MqttDeliveryToken deliveryToken = mqttTopic.publish(message);
-
-                logger.debug("Publishing message {} to topic '{}'", deliveryToken.getMessageId(), topic);
-                if (!async) {
-                    // wait for publish confirmation
-                    deliveryToken.waitForCompletion(10000);
-                    if (!deliveryToken.isComplete()) {
-                        logger.error(
-                                "Did not receive completion message within timeout limit whilst publishing to topic '{}'",
-                                topic);
+                MqttDeliveryToken deliveryToken;
+                try {
+                    deliveryToken = mqttTopic.publish(message);
+                    logger.debug("Publishing message {} to topic '{}'", deliveryToken.getMessageId(), topic);
+                    if (!async) {
+                        // wait for publish confirmation
+                        deliveryToken.waitForCompletion(10000);
+                        if (!deliveryToken.isComplete()) {
+                            logger.error(
+                                    "Did not receive completion message within timeout limit whilst publishing to topic '{}'",
+                                    topic);
+                        }
                     }
+                } catch (MqttException e) {
+                    logger.error("Could not publish message to topic {}", topic);
```

Comment:
I reduced it to trace but added the exception, because if publishing a message goes wrong this is probably only interesting for debug purposes and while debugging one is interested in the exception as well.